readthedocs/restapi/views/integrations.py

```diff
@@ -84,15 +110,26 @@ class GitHubWebhookView(WebhookMixin, APIView):
         }
     """
 
-    def handle_webhook(self, request, project, data=None):
+    def get_payload(self):
+        if self.request.content_type == 'application/x-www-form-urlencoded':
+            try:
+                return json.loads(self.request.data['payload'])
+            except ValueError:
+                pass
+
+    def handle_webhook(self):
+        data = self.get_payload()
+        if data is None:
+            data = self.request.data
```

Comment:
I can probably clarify this usage a bit more. I didn't want the payload method to override the default implementation unless neccessary -- that is, unless we had a case like GitHub sending a urlencoded form as `payload`.